### PR TITLE
PLYLoader: Ignore lines in the body

### PR DESCRIPTION
**Pull request by pobrn**
From the commit:
```
Instead of assuming that each line describes one object,
simply split the body at whitespaces and consume the resulting
token stream with no regard to line boundaries.

This is needed because, for example, some ply files exported
by librealsense use multiple lines to describe a single object[0].

[0]: https://github.com/IntelRealSense/librealsense/blob/e9f05c55f88f6876633bd59fd1cb3848da64b699/include/librealsense2/hpp/rs_export.hpp#L240
```

Side note: I have not seen much infrastructure for reporting errors, so premature EOF is simply silently ignored (and if there is a partial object, that is dropped).

---

<details>

<summary>librealsense code</summary>

```cpp
                for (size_t i = 0; i <new_verts.size(); ++i)
                {
                    out << new_verts[i].x << " ";
                    out << new_verts[i].y << " ";
                    out << new_verts[i].z << " ";
                    out << "\n";

                    if (mesh && use_normals)
                    {
                        out << normals[i].x << " ";
                        out << normals[i].y << " ";
                        out << normals[i].z << " ";
                        out << "\n";
                    }

                    if (use_texcoords)
                    {
                        out << unsigned(new_tex[i][0]) << " ";
                        out << unsigned(new_tex[i][1]) << " ";
                        out << unsigned(new_tex[i][2]) << " ";
                        out << "\n";
                    }
                }
```

</details>

<details>

<summary>sample from an exported file</summary>

```ply
ply
format ascii 1.0
comment pointcloud saved from Realsense Viewer
element vertex 163192
property float32 x
property float32 y
property float32 z
property float32 nx
property float32 ny
property float32 nz
property uchar red
property uchar green
property uchar blue
element face 311396
property list uchar int vertex_indices
end_header
-1.10254 0.844727 -1.55176
0.995526 0.0935393 -0.0133628
136 107 97
```

</details>

### CODE REVIEW

**Comment by Mugen87** [comment]
> This is needed because, for example, some ply files exported by librealsense use multiple lines to describe a single object[0].

Can you please share such PLY file in this PR?

**Comment by Mugen87** [comment]
So instead of:
```
-1.10254 0.844727 -1.55176 0.995526 0.0935393 -0.0133628 136 107 97
```
for defining vertex, normal and color it is:
```
-1.10254 0.844727 -1.55176
0.995526 0.0935393 -0.0133628
136 107 97
```
I can confirm that `PLYLoader` does currently not parse such files correctly.

I also could not find something in the unofficial specs that state such multi-line vertex data definitions are invalid. So I guess if meshlab can load the file it's appropriate to support this syntax as well. 

@mrdoob Are you okay with that?

File: examples/jsm/loaders/PLYLoader.js [hunk 1]
```diff
@@ -343,45 +347,53 @@ class PLYLoader extends Loader {
...
 			}
 
-			const lines = body.split( /\r\n|\r|\n/ );
-			let currentElement = 0;
-			let currentElementCount = 0;
-			let elementDesc = header.elements[ currentElement ];
-			let attributeMap = mapElementAttributes( elementDesc.properties );
-
-			for ( let i = 0; i < lines.length; i ++ ) {
+			class TokenStream {
```
Comment (Mugen87) [inline examples/jsm/loaders/PLYLoader.js:358]: Do you mind moving this class definition outside of `PLYLoader` into the module scope? Maybe right above the export statement...

File: examples/jsm/loaders/PLYLoader.js [hunk 2]
```diff
@@ -343,46 +345,31 @@ class PLYLoader extends Loader {
...
-			let currentElement = 0;
-			let currentElementCount = 0;
-			let elementDesc = header.elements[ currentElement ];
-			let attributeMap = mapElementAttributes( elementDesc.properties );
-
-			for ( let i = 0; i < lines.length; i ++ ) {
-
-				let line = lines[ i ];
-				line = line.trim();
-				if ( line === '' ) {
```
Comment (Mugen87) [inline examples/jsm/loaders/PLYLoader.js:364]: Since this code block has been removed: Is the loader still able to skip over empty lines?

**Comment by mrdoob** [comment]
> I also could not find something in the unofficial specs that state such multi-line vertex data definitions are invalid. So I guess if meshlab can load the file it's appropriate to support this syntax as well.
>
> @mrdoob Are you okay with that?

Sounds good 👍

**Review by Mugen87 — APPROVED** [review]
All test files work with this code as expected.

**Pull request merged**